#40 Implement Quick Outline

ghillairet opened the pull request.

ghillairet committed:
Update snakeyaml to 1.13, same version as in yedit.

ghillairet committed:
Update snakeyaml to 1.13, same version as in yedit.

ghillairet committed:
Update snakeyaml to 1.13, same version as in yedit.

ghillairet committed:
Update snakeyaml to 1.13, same version as in yedit.

ghillairet committed:
[#40] Implement Quick Outline

Prototype implementation

ghillairet committed:
[#40] Implement Quick Outline

Prototype implementation

ghillairet committed:
[#40] Implement Quick Outline

Prototype implementation

ghillairet committed:
[#40] Implement Quick Outline

Prototype implementation

ghillairet committed:
[#40] Implement Quick Outline

Prototype implementation

ghillairet committed:
Merge branch 'task/40' of github.com:RepreZen/SwagEdit into task/40

# Conflicts:
#	com.reprezen.swagedit/src/com/reprezen/swagedit/editor/SwaggerEditor.java

ghillairet committed:
[#40] Implement Quick Outline

ghillairet committed:
[#40] Implement Quick Outline

ghillairet committed:
[#40] Implement Quick Outline

ghillairet committed:
[#40] Implement Quick Outline

ghillairet committed:
Merge branch 'task/40' of github.com:RepreZen/SwagEdit into task/40

# Conflicts:
#	com.reprezen.swagedit/src/com/reprezen/swagedit/editor/SwaggerEditor.java

ghillairet committed:
Merge branch 'task/40' of github.com:RepreZen/SwagEdit into task/40

# Conflicts:
#	com.reprezen.swagedit/src/com/reprezen/swagedit/editor/SwaggerEditor.java
#	com.reprezen.swagedit/src/com/reprezen/swagedit/editor/outline/OutlineElement.java
#	com.reprezen.swagedit/src/com/reprezen/swagedit/editor/outline/OutlineStyledLabelProvider.java
#	com.reprezen.swagedit/src/com/reprezen/swagedit/editor/outline/QuickOutline.java

tfesenko commented on com.reprezen.swagedit/plugin.xml:
```diff
@@ -166,4 +167,41 @@
+         point="org.eclipse.ui.commands">
+      <command
+            description="Open Quick Outline"
+            id="com.reprezen.swagedit.keybinding.quickoutline"
```
Thank you for adding key binding for the command, except I am not sure why we need the word "keybinding" in commandId: `com.reprezen.swagedit.keybinding.quickoutline`

tfesenko commented on com.reprezen.swagedit/src/com/reprezen/swagedit/handlers/OpenQuickOutlineHandler.java:
```diff
@@ -0,0 +1,33 @@
+
+public class OpenQuickOutlineHandler extends AbstractHandler {
+
+    public static final int QUICK_OUTLINE = 513; // magic number from PDE quick outline see PDEProjectionViewer
```
Thank you for adding the comment..

tfesenko committed:
#169 - Empty swagger model (with  text) does not show an error in SwagEdit

tfesenko commented on com.reprezen.swagedit/.settings/org.eclipse.jdt.core.prefs:
```diff
@@ -1,7 +1,315 @@
 eclipse.preferences.version=1
 org.eclipse.jdt.core.compiler.codegen.inlineJsrBytecode=enabled
+org.eclipse.jdt.core.compiler.codegen.methodParameters=do not generate
```
@ghillairet is there any reason to keep settings in Git? We use Tycho to build it, so I don't see a valid reason  to share the settings, please let me know if you do.

ghillairet committed:
Remove settings that should not have been included.

ghillairet committed:
Revert changes made to snakeyaml version, will add it in separate pull request.

ghillairet committed:
[#40] Implement Quick Outline

ghillairet committed:
[#40] Implement Quick Outline

Fix id of command for quick outline
Fix tests, mockito cannot be used in xtend files

tfesenko commented on com.reprezen.swagedit.tests/src/com/reprezen/swagedit/editor/outline/OutlineElementTest.xtend:
```diff
@@ -0,0 +1,50 @@
+		assertEquals(2, el.children.size)
+		
+		assertEquals("hello", el.children.get(0).text)
+		assertEquals("world", el.children.get(1).text)
```
Thanks!

tfesenko commented on com.reprezen.swagedit/src/com/reprezen/swagedit/editor/SwaggerSourceViewerConfiguration.java:
```diff
@@ -83,8 +99,83 @@ public IReconciler getReconciler(ISourceViewer sourceViewer) {
+    @Override
+    public IInformationPresenter getInformationPresenter(ISourceViewer sourceViewer) {
+        // TODO Auto-generated method stub
+        return super.getInformationPresenter(sourceViewer);
```
Should we remove this default override?

tfesenko commented on com.reprezen.swagedit/src/com/reprezen/swagedit/editor/outline/QuickOutline.java:
```diff
@@ -0,0 +1,344 @@
+        return false;
+    }
+
+    public class NamePatternFilter extends ViewerFilter {
```
I wonder why a combination of `PatternFilter` + `FilteredTree` (instead of `Tree`) was not appropriate here. I am sure you had reasons to choose a `ViewerFilter`, just wondering why

ghillairet replied:
I could not make the quick outline look like the java quick outline with the filtertree. The java quickoutline doesn't use it for example.

tfesenko replied:
Makes sense

ghillairet committed:
Merge branch 'task/169' into task/40

# Conflicts:
#	com.reprezen.swagedit/src/com/reprezen/swagedit/editor/SwaggerEditor.java

tfesenko commented on com.reprezen.swagedit/src/com/reprezen/swagedit/editor/SwaggerEditor.java:
```diff
@@ -33,41 +33,58 @@
  * 
  */
-public class SwaggerEditor extends YEdit {
+public class SwaggerEditor extends YEdit implements IShowInSource, IShowInTarget {
```
`IShowInTarget` is used by`sourceViewerConfiguration`, but I did not figure out how `IShowInSource` is used by SwaggerEditor
